db/compaction/compaction_job.cc

```diff
@@ -567,6 +566,31 @@ void CompactionJob::GenSubcompactionBoundaries() {
   }
 }
 
+Status CompactionJob::FsyncOutputPaths() {
+  std::unordered_set<uint32_t> path_ids;
+
+  // iterate all output files and collect their path ids into a set
+  auto sub_compact_itr = compact_->sub_compact_states.begin();
+  for (; sub_compact_itr != compact_->sub_compact_states.end(); sub_compact_itr++) {
```

Comment:
Nit, can we use do `for (auto& sub_compact :  compact_->sub_compact_states)`?

Reply:
Fixed